File: 8632-879-724-find-pivot-index/8632-879-724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0,leftSum = 0,rightSum = 0;
        for(int i=0;i<nums.length;i++)
            sum+= nums[i];
        
        for(int i=0;i<nums.length;i++){
            rightSum = sum-leftSum;
            leftSum+= nums[i];
            if(rightSum==leftSum)
                return i;
        }
        return -1;
    }
}

File: 8632-879-724-find-pivot-index/8632-879-724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total,left,right = sum(nums),0,0

        for i in range(len(nums)):
            right = total-left
            left+= nums[i]
            if right==left:
                return i
        return -1